dev-docs/source/BuildingWithCMake.rst

```diff
@@ -0,0 +1,138 @@
+* Click "Configure" again and wait....
+* Finally, click "Generate". This will create the build files, e.g. there will be a Mantid.sln in the directory you selected as your build directory.
+
+Data Files Location
+-------------------
+
+Mantid used the CMake ExternalData system for managing testing data. See :ref:`DataFilesForTesting` for further instructions.
+
+With Qt Creator ...
+-------------------
+
+`Qt Creator <http://qt.nokia.com/products/developer-tools/>`_ has some really nice features (it's cross-platform, you can directly open Qt Designer within it, you can highlight a Qt type and go directly to it's help page, it knows about Qt types when debugging....).
+The nice feature in this context is that it has CMake support built in. So you can just open the project by pointing to the main CMakeLists file and then run CMake all within the IDE itself.
+
+Building and working with CMake
+###############################
+
+* You can now start your IDE and point to or import the generated solution/project files or run ``make``, ``nmake`` or ``jom`` to build the whole of Mantid (sub-targets are available - run ``make help`` to see them).
+* '''Visual Studio users''': Use the ``visual-studio.bat`` generated in the build directory to start the IDE. This sets up the environment correctly.
+* You should typically never have to run CMake manually again (unless you want to create a new, separate build) - it will be run automatically if one of the CMake input files changes.
+* It should be rare that you will need to edit the CMake build (``CMakeLists.txt``) files. The most common occurrence will be when you add a new file. This must be added to the corresponding CMakeLists file, e.g. if you add a file to Kernel, edit ``Mantid/Framework/Kernel/CMakeLists.txt`` to add the source, header and test files to the long lists of filepaths at the top of the file.
+* The class maker utility (:ref:`UsefulTools`) can edit the ``CMakeList.txt`` for you automatically
+* There are similar places in the Qt projects for ui files and files that need moc-ing.
+* If you add a new dependency, that will need to be added (this is less straightforward - do ask for help).
+* Cache variables can be added via the CMake Gui or by running ``ccmake``.
+
+Building the installer package
+##############################
+
+* For WIndows only, you first need to install NSIS, available at: http://nsis.sourceforge.net/Download. Ensure that the install directory is added to the PATH. You should be able to type ``makensis /?`` in a command prompt.
+* Run CMake with "ENABLE_CPACK" enabled. If using the GUI you need to click the "Advanced" checkbox to see this option.
+* You will now have a build target called "PACKAGE" available to create the installer package.
+
+Troubleshooting on OSX
+----------------------
```

Comment:
Is this relevant here? There is already a Troubleshooting section on the OS X specific page.

Reply:
This one is not specific to OSX, and relates to VATES. I can ask our local VATES developer to see if it's relevant enough to keep.